src/Compilers/CSharp/Portable/Binder/LocalScopeBinder.cs

```diff
@@ -226,6 +226,17 @@ protected ImmutableArray<LocalSymbol> BuildLocals(SyntaxList<StatementSyntax> st
                                 ExpressionVariableFinder.FindExpressionVariables(this, locals, vdecl, localDeclarationBinder);
                             }
 
+                            if (decl.Declaration.Type is ArrayTypeSyntax arrayTypeSyntax)
```

Comment:
>if (decl.Declaration.Type is ArrayTypeSyntax arrayTypeSyntax) [](start = 28, length = 61)

It is not clear why do we limit this to ```ArrayTypeSyntax```.

Reply:
I forgot to change that to visit subtypes which are arrays.

Reply:
> I forgot to change that to visit subtypes which are arrays.

Please add tests that expose the problem so that we could verify that the behavior is correct after you address the issue.

---
In reply to: [255648644](https://github.com/dotnet/roslyn/pull/32843#discussion_r255648644) [](ancestors = 255648644)